org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracingExtensions.xtend

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2017 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.generator.trace.node
+
+import java.util.function.Function
+import org.eclipse.emf.ecore.EFactory
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.EStructuralFeature
+import org.eclipse.xtend.lib.macro.AbstractClassProcessor
+import org.eclipse.xtend.lib.macro.Active
+import org.eclipse.xtend.lib.macro.TransformationContext
+import org.eclipse.xtend.lib.macro.declaration.InterfaceDeclaration
+import org.eclipse.xtend.lib.macro.declaration.MutableClassDeclaration
+import org.eclipse.xtend.lib.macro.declaration.ResolvedMethod
+import org.eclipse.xtext.generator.trace.ILocationData
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ */
+@Active(TracingExtensionsProcessor)
+annotation TracingExtensions {
```

Comment:
Maybe call this `TracedModelAccess`. I'm not in favor of calling this class `*Extension` as it actually is an active annotation. So technically `*ExtensionsProvider` would be correct ;)